From ffa531adf16d6de2362cc95f5776bc4667dab94f Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@toradex.com>
Date: Mon, 13 May 2019 17:56:55 +0300
Subject: [PATCH 1/7] imx8: cpu: fix warning for cpu_imx_get_temp
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

cpu_imx_get_temp() definition is wrapped with a ifdef macro, which leads
to warnings if CONFIG_IMX_SCU_THERMAL isn't defined and there are still
references to this function, as, for example, in cpu_imx_get_desc().
Drop ifdef as linker will automatically remove the function
in case CONFIG_IMX_SCU_THERMAL is not enabled.

Fix warning:
arch/arm/mach-imx/imx8/cpu.c: In function ‘cpu_imx_get_desc’:
arch/arm/mach-imx/imx8/cpu.c:612:40: warning: implicit declaration of
function ‘cpu_imx_get_temp’; did you mean ‘cpu_imx_get_desc’?
[-Wimplicit-function-declaration]
    ret = snprintf(buf, size, " at %dC", cpu_imx_get_temp());
                                         ^~~~~~~~~~~~~~~~
                                         cpu_imx_get_desc
                                         cpu_imx_get_desc

Fixes: 82467cb217 ("imx8: cpu: get temperature when print cpu desc")
Signed-off-by: Igor Opaniuk <igor.opaniuk@toradex.com>
---
 arch/arm/mach-imx/imx8/cpu.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/arch/arm/mach-imx/imx8/cpu.c b/arch/arm/mach-imx/imx8/cpu.c
index 4eb15de93e..d30ffcde3a 100644
--- a/arch/arm/mach-imx/imx8/cpu.c
+++ b/arch/arm/mach-imx/imx8/cpu.c
@@ -815,7 +815,6 @@ const char *get_core_name(void)
 		return "?";
 }
 
-#if IS_ENABLED(CONFIG_IMX_SCU_THERMAL)
 static int cpu_imx_get_temp(void)
 {
 	struct udevice *thermal_dev;
@@ -834,12 +833,6 @@ static int cpu_imx_get_temp(void)
 
 	return cpu_tmp;
 }
-#else
-static int cpu_imx_get_temp(void)
-{
-	return 0;
-}
-#endif
 
 int cpu_imx_get_desc(struct udevice *dev, char *buf, int size)
 {
-- 
2.13.6

